libdrgn/dwarf_info_cache.c

```diff
@@ -42,6 +42,53 @@ static bool drgn_type_realloc(struct drgn_type **type, size_t capacity,
 	return true;
 }
 
+/*
+ * We use dwarf_formudata() to pull constants out of dwarf attributes.
+ * Unfortunately, GCC's dwarfv2 generator for structure member offsets uses
+ * simple "blocks" instead of the constants that dwarf_formudata() expects.
+ *
+ * So if dwarf_formudata() fails, we handle the special case of the attribute
+ * being a block consisting of a single DW_OP_plus_uconst operation.
+ */
+static int drgn_formudata(Dwarf_Attribute *attr, Dwarf_Word *return_uval)
```

Comment:
How about naming this `drgn_dwarf_uconst()` to separate its naming from any particular `DW_FORM`?

Reply:
Yeah, this would be fine.